### PR TITLE
add ability to interpret json schema format

### PR DESCRIPTION
**Pull request by tharwan**
for our use case it would be very useful to have date and time pickers available. Since Draft 7 of json schema allows for [specific formats](https://json-schema.org/understanding-json-schema/reference/string#dates-and-times) in the schema, I thought it might be nice to add ability to interpret the format specification for date-time, date and time.

Any suggestions to improve the implementation are highly welcome.

### CODE REVIEW

File: ipywidgets_jsonschema/form.py [hunk 1]
```diff
@@ -197,6 +211,21 @@ def _construct(self, schema, label=None, root=False):
...
             raise FormError("Not accepting arrays of types currently")
+
+        format_ = schema.get("format", None)
+        available_fmt = []
+        if hasattr(ipywidgets, "DatetimePicker"):
+            available_fmt.append("date-time")
+        if hasattr(ipywidgets, "DatePicker"):
+            available_fmt.append("date")
+        if hasattr(ipywidgets, "TimePicker"):
+            available_fmt.append("time")
```
Comment (dokempf) [inline ipywidgets_jsonschema/form.py:222]: What is the rationale behind these checks? Have the relevant widgets been introduced very recently into `ipywidgets`?

**Review by dokempf — CHANGES_REQUESTED** [review]
Thanks for providing this. I think it makes a great addition to the library. I added one question as a review comment but would merge and release soon after that is clarified.

File: ipywidgets_jsonschema/form.py (hunk 1 above)
Comment (tharwan) [reply ipywidgets_jsonschema/form.py:222]: Yes, they were added in 8. The test suite also runs agains 7, so in order to pass the tests I had to add these checks.

Comment (dokempf) [reply ipywidgets_jsonschema/form.py:222]: Ah yes, I have slightly rewritten this in #87, as I would like to avoid doing to much `hasattr` stuff.

**Pull request merged**